cmd/gaia/init/validate_genesis.go

```diff
@@ -0,0 +1,48 @@
+package init
+
+import (
+	"fmt"
+
+	"github.com/cosmos/cosmos-sdk/cmd/gaia/app"
+	"github.com/cosmos/cosmos-sdk/codec"
+	"github.com/cosmos/cosmos-sdk/server"
+	"github.com/pkg/errors"
+	"github.com/spf13/cobra"
+	"github.com/tendermint/tendermint/types"
+)
+
+// Validate genesis command takes
+func ValidateGenesisCmd(ctx *server.Context, cdc *codec.Codec) *cobra.Command {
+	return &cobra.Command{
+		Use:   "validate-genesis [file]",
+		Args:  cobra.RangeArgs(0, 1),
+		Short: "validates the genesis file at the default location or at the location passed as an arg",
+		RunE: func(cmd *cobra.Command, args []string) (err error) {
+			var genesis string
+			var genDoc types.GenesisDoc
+			var genstate app.GenesisState
+
+			// Load default if passed no args, otherwise load passed file
+			if len(args) == 0 {
+				genesis = ctx.Config.GenesisFile()
+			} else {
+				genesis = args[0]
+			}
+
```

Comment:
I'd add a diagnostic info here just to inform the user which location `genesis.json` is being read from:

```
fmt.Fprintf(os.Stderr, "validating %s", genesis)
```

Reply:
Great add! Thank you!